netlogo-gui/src/main/app/ChangeManager.scala

```diff
@@ -0,0 +1,138 @@
+package org.nlogo.app
+
+import java.util.List
+import java.util.ArrayList
+import javax.swing.event.ChangeEvent
+/**
+  * Manages a Queue of Changables to perform undo and/or redo operations. Clients can add implementations of the Changeable
+  * class to this class, and it will manage undo/redo as a Queue.
+  *
+  * @author Greg Cope
```

Comment:
OK, the copyright is definitely not acceptable for use in NetLogo. We need to remove this file immediately. You can see whether `UndoManager` will work, perhaps it's generalizable enough to fit here. If it won't we can write something with similar behavior to this, but it needs to be original code written without reference to this code.